### ISSUE TITLE
VAST Error: on VASTClient.buildVastTree

### ISSUE THREAD

**Issue by BeastModeON** [MailOnline/videojs-vast-vpaid#20]
Hi! Getting this error on any VAST/VPAID XML when I try to play Ad: 
```
VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) The video could not be loaded, either because the server or network failed or because the format is not supported
```
Console output:
```
AD ERROR: VAST Error: on VASTClient.buildVastTree, error parsing xml z {message: "VAST Error: on VASTClient.buildVastTree, error parsing xml", code: 100}code: 100message: "VAST Error: on VASTClient.buildVastTree, error parsing xml"stack: (...)__proto__: Error
    at http://MYDOMAIN.com/videojs/vpaid/videojs-vast-vpaid.min.js:2:29160
    at http://MYDOMAIN.com/videojs/vpaid/videojs-vast-vpaid.min.js:3:6281name: "VAST Error"stack: (...)get stack: ()arguments: nullcaller: nulllength: 0name: ""prototype: aK__proto__: ()<function scope>No Scopesset stack: ()arguments: nullcaller: nulllength: 1name: ""prototype: aN__proto__: ()<function scope>__proto__: DefineError.aTconstructor: Error()message: ""name: "Error"toString: toString()__proto__: Object__defineGetter__: __defineGetter__()__defineSetter__: __defineSetter__()__lookupGetter__: __lookupGetter__()__lookupSetter__: __lookupSetter__()constructor: Object()hasOwnProperty: hasOwnProperty()isPrototypeOf: isPrototypeOf()propertyIsEnumerable: propertyIsEnumerable()toLocaleString: toLocaleString()toString: toString()valueOf: valueOf()get __proto__: get __proto__()set __proto__: set __proto__() undefined
```

Any ideas how to fix this?

**Comment by Fetz** [comment]
Hi, can you give us the vast request that is giving that error so we can test?

Thank you

**Comment by LastStarfi9hter** [comment]
firstly, great plugin. blows the others away.

I've encountered this too with url
http://ad5.liverail.com/?LR_PUBLISHER_ID=58163&LR_SCHEMA=vast2

Cheers.

**Comment by BeastModeON** [comment]
Actually I did solve the problem.
First - I was linked to outdated version of VideoJS.
Second - I've debeautified all min.js.
Now it's seems to be working.

**Comment by Fetz** [comment]
@BeastModeON thanks, what version of videojs you were using? maybe will be cool to update the readme, saying what version of videojs we support.

@LastStarfi9hter thanks ^^ for the complement. Regarding the link in  MAC Chrome (44.0.2403.130) is working but in firefox 39 is not working because the request is being blocked by the browsers because they aren't [sending the headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS) or [VAST IAB specs] (http://www.iab.net/media/file/VAST_3.0_Draft_04-10-2012.pdf) 2.1.5.2 - page 16,but sometimes some providers they don't follow the specification fully.

Some players don't have this issue because they are loading the vast request through Flash ([the Flash CORS, it doesn't use the header of the request, they use a cross-domain.xml] (http://tiffanybbrown.com/2011/10/10/html5-for-as3-developers-cross-domain-xml-and-cross-origin-resource-sharing/)).

If you have issues with some providers, ask them to enable the headers. Because more and more flash will be less important (some browsers are blocking flash, or will block).

**Comment by Fetz** [comment]
closing the issue since the main issue was an issue with old version of video.js. @LastStarfi9hter if you have still some issue with that ad or another please create a new issue.
